src/opencl_rawsha512_gpl_fmt_plug.c

```diff
@@ -197,145 +246,56 @@ static void create_clobj(size_t gws, struct fmt_main *self)
 		mask_cand = mask_int_cand.num_int_cand;
 		mask_gws = gws;
 	}
+	// create pinned buffers
+	opencl_create_map(
+		CL_MEM_READ_ONLY | CL_MEM_ALLOC_HOST_PTR, BUFFER_SIZE * gws,
+		&pinned_plaintext,     /* buffer created on device */
+		(void *) &plaintext);  /* mapped buffer created on host */
```

Comment:
(Many places)

No need to cast to `void*`, that's sort of the idea with using it as function parameter!

Reply:
```
opencl_rawsha512_gpl_fmt_plug.c:255: warning: passing argument 4 of ‘opencl_create_map’ from incompatible pointer type
opencl_common.h:306: note: expected ‘void **’ but argument is of type ‘uint32_t **’

```

Reply:
Oh I didn't know that, pardon me. Apparently it's not allowed with a double pointer. I can't fathom why.